cuda/base/cusparse_bindings.hpp

```diff
@@ -629,6 +704,49 @@ inline void destroy(csrgemm2Info_t info)
 }
 
 
+#if defined(CUDA_VERSION) && (CUDA_VERSION >= 11000)
+
+
+inline cusparseDnVecDescr_t create_dnvec(int64_t size, void *values,
```

Comment:
Suggestion: How about replacing void* values by a template parameter and inferring `valueType = cuda_data_type<ValueType>`? Same for create_csr